pkg/client/restclient/config.go

```diff
@@ -130,9 +130,15 @@ type ContentConfig struct {
 	// a RESTClient directly. When initializing a Client, will be set with the default
 	// code version.
 	GroupVersion *unversioned.GroupVersion
+	// NegotiatedSerializer is used for obtaining encoders and decoders for multiple
+	// supported media types.
+	NegotiatedSerializer runtime.NegotiatedSerializer
+
 	// Codec specifies the encoding and decoding behavior for runtime.Objects passed
 	// to a RESTClient or Client. Required when initializing a RESTClient, optional
 	// when initializing a Client.
+	//
+	// DEPRECATED: Please use NegotiatedSerializer instead.
```

Comment:
If this takes precedence (or does not) the doc should specify its relationship to NegotiatedSerializer.  I suspect there are still cases where you want to pass a codec and not do negotiation, so I don't know that this is deprecated.